### ISSUE TITLE
Move snapshots to plugin

### ISSUE THREAD

**Issue by krzkaczor** [l2beat/earl#106]
Not everyone uses snapshots but the current implementation comes with many dependencies so it makes sense to extract it to a separate plugin.

@sz-piotr you're gonna love it :->

**Comment by krzkaczor** [comment]
I agree that one of the earl's strength is having the "batteries" included. I just don't believe that snapshots are critical enough to justify having them in the core. 

You see, everyone uses mocks and basic matchers shipped with earl and hence it makes sense to ship with them. Using snapshots seems to be much more opinionated. Personally, I don't use them at all. I believe neither do you. Of course, there are people out there longing for this feature and that's why I added them in the first place. However, I think we can safely extract it from the core as long as:

- It's painless to enable them 
- It won't affect the quality of the feature

Basically, it's the same reasoning as for not having blockchain-related features in earl's core... 

Just to be clear: what I am planning to do is making snapshots an official plugin. It will be part of this monorepo, with integration tests testing it against the newest earl.

**Comment by sz-piotr** [comment]
Bump. Changed my mind, no point re-implementing, this should be a plugin

**Comment by sz-piotr** [comment]
Welp, ended up re-implementing snapshots. Oh well, turns out it was *easy*
